cli/command/container/run_test.go

```diff
@@ -61,42 +128,60 @@ func TestRunAttachTermination(t *testing.T) {
 		},
 		containerAttachFunc: func(ctx context.Context, containerID string, options container.AttachOptions) (types.HijackedResponse, error) {
 			server, client := net.Pipe()
+			conn = server
 			t.Cleanup(func() {
 				_ = server.Close()
 			})
 			attachCh <- struct{}{}
 			return types.NewHijackedResponse(client, types.MediaTypeRawStream), nil
 		},
+		waitFunc: func(_ string) (<-chan container.WaitResponse, <-chan error) {
+			responseChan := make(chan container.WaitResponse, 1)
+			errChan := make(chan error)
+
+			responseChan <- container.WaitResponse{
+				StatusCode: 130,
+			}
+			return responseChan, errChan
+		},
 		Version: "1.36",
 	}, func(fc *test.FakeCli) {
 		fc.SetOut(streams.NewOut(tty))
 		fc.SetIn(streams.NewIn(tty))
 	})
-	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGTERM)
-	defer cancel()
-
-	assert.Equal(t, fakeCLI.In().IsTerminal(), true)
-	assert.Equal(t, fakeCLI.Out().IsTerminal(), true)
 
 	cmd := NewRunCommand(fakeCLI)
 	cmd.SetArgs([]string{"-it", "busybox"})
 	cmd.SilenceUsage = true
+	cmdErrC := make(chan error, 1)
 	go func() {
-		assert.ErrorIs(t, cmd.ExecuteContext(ctx), context.Canceled)
+		cmdErrC <- cmd.Execute()
 	}()
 
+	// run command should attempt to attach to the container
 	select {
 	case <-time.After(5 * time.Second):
-		t.Fatal("containerAttachFunc was not called before the 5 second timeout")
+		t.Fatal("containerAttachFunc was not called before the timeout")
 	case <-attachCh:
 	}
 
-	assert.NilError(t, syscall.Kill(syscall.Getpid(), syscall.SIGTERM))
+	syscall.Kill(syscall.Getpid(), syscall.SIGINT)
```

Comment:
Ugh, I know it was there already, but I'm not thrilled about sending signals to the running process. 🙈 
This feels like it should be a e2e test and not an integration one, as it does have actual side effects to the whole process running tests (which may also affect other tests in case of failures/parallel tests running).

I'm _okay_ with this as an immediate fix, but we should really turn these into an e2e test.

Reply:
Heh, was actually looking at that as well; it felt more like this could be / should be an e2e test 🤔

Reply:
I agree, but I think (for now) this makes sense – or rather, without testing this functionality, this test doesn't really make sense (and I did consider just deleting it/covering it with an e2e test – we already have a similar one in https://github.com/docker/cli/blob/826fc32e82e23bb5f80e85d8777427c5f0c24b4d/e2e/container/attach_test.go).

This isn't the overall CLI signal handling we're testing, but the specific run handling.

Reply:
Yes, I'm good with keeping it as-is for now; we could have a look at cleaning up some of this at some point

Reply:
I'm on it, I just want to get this merged so that my follow ups aren't running into this every other time. This test has been failing a lot and I just want to fix it 😭